src/components/Candidates/CandidateForm.vue

```diff
@@ -0,0 +1,162 @@
+<script setup>
+import { ref } from "vue";
+import { Form, Field } from "vee-validate";
+import { v4 as uuidv4 } from "uuid";
+
+const candidateData = ref({
+  position: "",
+  username: "",
+  linkedinUrl: "",
+  feedback: "",
+  avatarUrl: "",
+  id: 0,
+});
+
+const candidateSchema = {
+  position(value) {
+    if (!value) {
+      return "Position field is required";
+    }
+    if (value.length <= 5) {
+      return "Position has to be bigger than 5 chars";
+    }
+    return true;
+  },
+  username(value) {
+    if (!value) {
+      return "Username field is required";
+    }
+    if (value.length <= 5) {
+      return "Username has to be bigger than 5 chars";
+    }
+    return true;
+  },
+  feedback(value) {
+    if (value && value.length > 150) {
+      return "Feedback max length 150 chars";
+    }
+    return true;
+  },
+};
+
+function onSubmit(values, { resetForm }) {
+  candidateData.value = { ...values };
+  candidateData.value.id = uuidv4();
+  console.log(candidateData.value);
+  // TODO:send candidateData to mockAPI, test api call
+  resetForm();
+}
+</script>
+<template>
+  <div>
+    <button
+      type="button"
+      class="btn btn-primary"
+      data-bs-toggle="modal"
+      data-bs-target="#exampleModal"
+    >
+      Add candidate
+    </button>
+
+    <!-- Modal -->
```

Comment:
Use `b-modal` component
https://bootstrap-vue.org/docs/components/modal#modal

I'm just wondering if in future we will decide to add more fields on this form modal would be best solution in this case?